### ISSUE TITLE
Add browser restart capability

### ISSUE THREAD

**Issue by rpappalax** [mozilla/FoxPuppet#124]
For shavar-e2e-tests, we'll need not only to set preferences, but to restart the browser with a pre-existing profile (in a specified path), for the preferences to take effect in Firefox

**Comment by jrbenny35** [comment]
Digging around I found that the developer console allows for sending the command ```restart``` and the browser restarting. Maybe that could work.

**Comment by whimboo** [comment]
@jrbenny35, as I have explained on IRC yesterday, what you want to do is creating end-points for those commands in geckodriver, so that it can be forwarded to Marionette which implements this correctly. Don't do restarts without letting geckodriver know about it.

**Comment by davehunt** [comment]
We can already provide a profile path when starting Firefox, would this avoid the need for a restart?

**Comment by jrbenny35** [comment]
I guess this is more of a geckodriver thing.

**Comment by rpappalax** [comment]
@davehunt no, we need to set the path for the binary and profile, but the real issue here is having the browser restart after setting some custom prefs for shavar testing.  these prefs don't activate the the tracking protection file downloads until the browser has been restarted.

**Comment by davehunt** [comment]
> custom prefs for shavar testing

Do you have some more details? We should be able to put any preferences in place before the initial browser launch, which would replicate restarting after changing these preferences.

**Comment by rpappalax** [comment]
@davehunt perhaps i can do a walkthrough with you and @jrbenny35 but we need to
1. set prefs
2. start browser
3. quit browser
4. start browser for actual test

**Comment by davehunt** [comment]
@rpappalax a walk-through would be great, as I'm almost certainly missing something 😄

**Comment by whimboo** [comment]
I don't really see why steps 3 and 4 are necessary when the prefs are set in 1, and no other pref is set between 2 and 3. So starting the browser in 2 should already have this preference enabled.
